src/components/QueryEditor.tsx

```diff
@@ -1,59 +1,31 @@
 import React, { PureComponent } from 'react';
 import defaults from 'lodash/defaults';
-import { BracesPlugin, QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
+import { QueryField, Select, Field, Input, Collapse } from '@grafana/ui';
 import { QueryEditorProps, SelectableValue } from '@grafana/data';
-import { DataSource } from './datasource';
-import { defaultQuery, LightstepDataSourceOptions, LightstepQuery } from './types';
+import { DataSource } from '../datasource';
+import { LightstepDataSourceOptions, LightstepQuery } from '../types';
 
 type Props = QueryEditorProps<DataSource, LightstepQuery, LightstepDataSourceOptions>;
-interface QueryEditorState {
-  errorMessage: string;
-  projects: string[];
+type State = {
   isOptionsOpen: boolean;
-}
+};
 
-export class QueryEditor extends PureComponent<Props, QueryEditorState> {
-  plugins: Plugin[] = [BracesPlugin()];
-  state: QueryEditorState = {
+/**
+ * Component responsible for the query text and options editing UI.
+ */
+export class QueryEditor extends PureComponent<Props, State> {
+  state = {
     isOptionsOpen: false,
-    errorMessage: '',
-    projects: [],
   };
 
-  componentDidMount() {
-    this.setState({ projects: this.props.datasource.fetchProjects() });
-  }
+  onQueryChange = (value: string) => {
+    const { onChange, query } = this.props;
 
-  componentWillUnmount() {
-    // Refresh chart if query is deleted
-    if (this.props.onRunQuery) {
-      this.props.onRunQuery();
-    }
-  }
```

Comment:
This comment is a bit confusing, and the only way this can happen is if the user changes the query, and then clicks the back to dashboard button without running the query. This seems extremely unlikely since they wouldn't know if their query worked successfully. Removing since there isn't value added to simplify the code paths this component is responsible for managing.